### PR TITLE
WF-335 Add Transactions Journey

### PR DESCRIPTION
**Pull request by kirillsud**
Implemented **Transactions List Journey** that displays a list of transactions from mock data and allows filter them. Also the journey is attached to the route `transactions` and added a menu item that points to this route.

### CODE REVIEW

File: .npmrc [hunk 1]
```diff
@@ -1 +1 @@
-@backbase:registry=https://artifacts.backbase.com/api/npm/npm-backbase/
+@backbase:registry=https://repo.backbase.com/api/npm/npm-backbase/
```
Comment (santam85) [inline .npmrc:1]: Nice one thanks

**Review by santam85 — APPROVED** [review]

File: projects/transactions-journey/src/lib/pipes/filter-transactions.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export class FilterTransactionsPipe implements PipeTransform {
+
+  transform(value: Transaction[], text: string): Transaction[] {
+    if (!text) {
+      return value;
+    }
+
+    text = text.toLocaleLowerCase();
+
+    return value.filter(x =>
```
[the remark this answers is not in the document]
Comment (georgejolema) [reply projects/transactions-journey/src/lib/pipes/filter-transactions.pipe.ts]: done

File: projects/transactions-journey/src/lib/pipes/filter-transactions.pipe.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  name: 'filterTransactions'
+})
+export class FilterTransactionsPipe implements PipeTransform {
+
+  transform(value: Transaction[], text: string): Transaction[] {
+    if (!text) {
+      return value;
+    }
+
+    text = text.toLocaleLowerCase();
```
[the remark this answers is not in the document]
Comment (georgejolema) [reply projects/transactions-journey/src/lib/pipes/filter-transactions.pipe.ts]: fixed

File: projects/transactions-journey/src/lib/views/transactions-view/transactions-view.component.scss [hunk 4]
```diff
@@ -0,0 +1,18 @@
+ul {
```
[the remark this answers is not in the document]
Comment (georgejolema) [reply projects/transactions-journey/src/lib/views/transactions-view/transactions-view.component.scss]: I removed the css file completely and used css classes instead

File: projects/transactions-journey/src/lib/transactions-journey.service.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { Transaction } from './model/transaction';
+import * as mocks from './mock-data/transactions.json';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TransactionsJourneyService {
```
[the remark this answers is not in the document]
Comment (georgejolema) [reply projects/transactions-journey/src/lib/transactions-journey.service.ts]: I renamed the service and applied interceptor. Let me know what you think

**Comment by yenobi** [comment]
I resolved the conversation about the interceptor, so will just leave a comment - nice, like that you have agreed 👍

**Review by yenobi — APPROVED** [review]

File: projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<span class="categoryCode" [ngStyle]="{'background-color': transaction.categoryCode}"></span>
+<span class="date">{{ transaction.dates.valueDate | date: 'MMM. d' }}</span>
+<span class="description">
+  <span class="merchant">{{ transaction.merchant.name }}</span>
```
Comment (jesus-crysist) [inline projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.html:4]: Shall we follow BEM methodology for naming classes here as proposed by Design System team? Or it is not necessary?

File: projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.spec.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    .compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(TransactionItemComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (jesus-crysist) [inline projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.spec.ts:22]: Would you like to add more unit tests to cover multiple ways to test component? For example, simple unit test, and shallow component testing.

File: projects/transactions-journey/src/lib/pipes/filter-transactions.pipe.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { Transaction } from '../model/transaction';
+
+@Pipe({
+  name: 'filterTransactions'
+})
+export class FilterTransactionsPipe implements PipeTransform {
+
+  transform(value: Transaction[], text: string): Transaction[] {
```
Comment (jesus-crysist) [inline projects/transactions-journey/src/lib/pipes/filter-transactions.pipe.ts:9]: Should there be a unit test for this pipe?

**Review by jesus-crysist — APPROVED** [review]

File: projects/transactions-journey/src/lib/pipes/filter-transactions.pipe.ts (hunk 8 above)
Comment (georgejolema) [reply projects/transactions-journey/src/lib/pipes/filter-transactions.pipe.ts:9]: I will add unit tests as part of the scope of this ticket if you don't mind https://backbase.atlassian.net/browse/WF-336

File: projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.spec.ts (hunk 7 above)
Comment (georgejolema) [reply projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.spec.ts:22]: I will add unit tests as part of the scope of this ticket if you don't mind https://backbase.atlassian.net/browse/WF-336

File: projects/transactions-journey/src/lib/transactions.http.service.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TransactionsHttpService {
+  constructor(private http: HttpClient) {}
+
+  getTransactions(page: number, pageSize: number): Observable<Transaction[]> {
+    const params = new HttpParams()
+      .append('page', page + '')
```
[the remark this answers is not in the document]
Comment (georgejolema) [reply projects/transactions-journey/src/lib/transactions.http.service.ts]: done

File: src/app/services/transactions-interceptor.ts [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HttpResponse } from '@angular/common/http';
+import * as mocks from '../mock-data/transactions.json';
+import { Transaction } from 'transactions-journey';
+import { Observable, of } from 'rxjs';
+
+@Injectable()
+export class TransactionsInterceptor implements HttpInterceptor {
+  intercept(req: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    if (req.url.endsWith('api/transactions')) {
+      const page = req.params.get('page') || 0;
```
[the remark this answers is not in the document]
Comment (georgejolema) [reply src/app/services/transactions-interceptor.ts]: done

File: projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.html (hunk 6 above)
Comment (georgejolema) [reply projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.html]: done

File: projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.spec.ts (hunk 7 above)
Comment (jesus-crysist) [reply projects/transactions-journey/src/lib/components/transaction-item/transaction-item.component.spec.ts:22]: Sure. 👍

**Pull request merged**